exec/java-exec/src/main/java/org/apache/drill/exec/planner/physical/JoinPruleBase.java

```diff
@@ -246,4 +246,17 @@ public RelNode convertChild(final DrillJoinRel join,  final RelNode rel) throws
 
   }
 
+  // DRILL-6089 make sure no collations are added to HashJoin
+  public static RelTraitSet removeCollation(RelTraitSet traitSet, RelOptRuleCall call)
+  {
+    RelTraitSet newTraitSet = call.getPlanner().emptyTraitSet();
+
+    for (RelTrait trait: traitSet) {
+      if (!trait.getTraitDef().getTraitClass().equals(RelCollation.class)) {
```

Comment:
I suppose you already considered Calcite's RelTraitSet.replace() method and found this to be simpler ?

Reply:
I don't see a way to remove a trait with RelTraitSet.replace(). So I made this removeCollation method instead.